apps/frontend/src/hooks/aave/useAaveRates.tsx

```diff
@@ -0,0 +1,300 @@
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getBaseVariableBorrowRate',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getMaxVariableBorrowRate',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getVariableRateSlope1',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getVariableRateSlope2',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+export interface RatesDataResult {
+  currentUsageRatio: string;
+  optimalUsageRatio: string;
+  baseVariableBorrowRate: string;
+  variableRateSlope1: string;
+  variableRateSlope2: string;
+  underlyingAsset: string;
+  name: string;
+  symbol: string;
+  decimals: string;
+}
+
+const calculateUtilizationRate = (
+  decimals: number,
+  totalDebt: string,
+  availableLiquidity: string,
+): bigint => {
+  // Create BigNumber instances
+  const totalBorrow = BigInt(utils.parseUnits(totalDebt, decimals).toString());
+  const totalSupply = BigInt(availableLiquidity) + totalBorrow;
+  // Perform division
+  return (totalBorrow * BigInt(10 ** 18)) / totalSupply;
+};
+
+export const useAaveInterestRatesData = (): {
+  data: RatesDataResult | null;
+  error: string | null;
+} => {
+  const [data, setData] = useState<RatesDataResult | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const { provider } = useAccount();
+
+  const [searchParams] = useSearchParams();
+  const symbol = searchParams.get('asset') || 'ETH';
+  const { reserves, loading } = useAaveReservesData();
+  const reserve: Reserve | undefined = reserves.find(
+    r => r.symbol.toLocaleLowerCase() === symbol.toLocaleLowerCase(),
+  );
+  const interestRateStrategyAddress = reserve?.interestRateStrategyAddress;
+
+  const rateContract = useMemo(
+    () =>
+      provider && interestRateStrategyAddress && !loading
+        ? new Contract(
+            interestRateStrategyAddress,
+            INTEREST_RATE_STRATEGY_ABI,
+            provider,
+          )
+        : null,
+    [loading, interestRateStrategyAddress, provider],
+  );
+
+  useEffect(() => {
+    if (loading || !rateContract || !reserve) return;
+    let ratesData: RatesDataResult;
```

Comment:
let's fo `const ratesData = ...` in line 267. Not a variable and not being used outside of try catch so...